### PR TITLE
Fix issue where non-standard media types were not showing requestData

### PR DESCRIPTION
**Pull request by baynezy**
Previously, perfectly valid media types that are JSON but not an
exact match for 'application/json' e.g 'application/json; charset=charset=utf-8'
were causing requestData not to show

### CODE REVIEW

File: lib/helpers.js [hunk 1]
```diff
@@ -60,28 +62,36 @@ function validateResponse(type, noSchema,
...
 }
 
 /**
  * determines if the mediatype is json
  * @param {string} type   content type to be evaluated
  */
-function isJsonMediaType(type) {
-  return type.trim().toLowerCase().substr(type.length - 4) === 'json';
+function mediaTypeContainsJson(type) {
+  return /\bjson\b/.test(type);
```
Comment (Conorrr) [inline lib/helpers.js]: This should be case-insensitive

Comment (baynezy) [inline lib/helpers.js]: Thanks Conor. I'll fix that.

File: test/request-data/test.js [hunk 2]
```diff
@@ -1,27 +1,3 @@
...
- * all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
- * THE SOFTWARE.
- */
```
Comment (noahdietz) [inline test/request-data/test.js]: Why are we removing the license from this file?

**Comment by noahdietz** [comment]
LGTM, except one comment, happy to merge after it is addressed

**Comment by baynezy** [comment]
@noahdietz my bad I'll add that back.

**Comment by baynezy** [comment]
@noahdietz - any danger of a minor NPM release with this in?

**Comment by noahdietz** [comment]
@baynezy I made a `patch` release: `1.4.2`

**Comment by baynezy** [comment]
Thanks so much @noahdietz

**Pull request merged**